rclcpp/src/rclcpp/node_options.cpp

```diff
@@ -266,6 +268,20 @@ NodeOptions::parameter_event_qos(const rclcpp::QoS & parameter_event_qos)
   return *this;
 }
 
+const rclcpp::QoS &
+NodeOptions::rosout_qos() const
+{
+  return this->rosout_qos_;
+}
+
+NodeOptions &
+NodeOptions::rosout_qos(const rclcpp::QoS & rosout_qos)
+{
+  this->node_options_.reset();
+  this->rosout_qos_ = rosout_qos;
+  return *this;
+}
```

Comment:
I am not so convinced that we should provide getter/sertter. since QoS is static profile for the endpoints, it would be better to align with [parameter_event_qos_](https://github.com/ros2/rclcpp/blob/f2cd2fbf0e2737a94585666dd6c4eeffaa3b50e9/rclcpp/include/rclcpp/node_options.hpp#L357-L359)

Reply:
> I am not so convinced that we should provide getter/sertter. 

Even if consisent with `parameter_event_qos_`, the getter/setter interface is also neccessary. 

The user need to set custom qos profile like this:
```
Node("talker", options.rosout_qos(rclcpp::QoS(rclcpp::KeepLast(1000)).transient_local().XXX))
```

Reply:
sorry never mind this one.